docs/dev/migrations.md

```diff
@@ -19,4 +19,12 @@ Migrations are run at startup right after the automigration is run.
 - A new migration scaffold can be created using `MIGRATION_NAME=someMigrationNameHere make generate-migration`
     - This will give the migration files a proper timestamp as well as (empty) tests
 
-To be run, every migration function should be added to the list in `migrations.all()`
+To be run, every migration function should be added to the list in `postMigrations`
+
+### Pre-migrations
+
+We currently have a single migration in `preMigrations`, i.e. running before `AutoMigrate` step that is responsible for synchronizing the database schema.
+
+It is a bad decision, but given that it has already been rolled out, it would be extremely difficult to move it again to `postMigrations` with the guarantee that no deployment is going to break or end up in a state that is difficult to recover.
```

Comment:
I think rewording this would be better.

IIUC running this before the automigrate was necessary, so I don't think "bad decision" is fair.
Maybe here is a good place to document the reasoning behind why we had to run this migration before auto migrate even if it isn't exactly the best situation.

From a slack conversation at the time I have this info:
> because the automigrate will fail in this case unless a column will be created and populated first since we added to the model a column that will be part of the PK and that cannot be NULL

Since Ravid left and didn't put any info in the PR or commit can @avishayt or @yevgeny-shnaidman maybe clear up the reasoning for this?

Reply:
I completely agree that it was probably the best solution back then (if not the only possible). But as you said, today we don’t seem to have the knowledge on why it’s been done like this and we want to discourage people from doing the same in the future. So yeah, I do hope someone in the team has the knowledge we could write here in the doc

Reply:
```suggestion
The use of `preMigrations` is (currently) discouraged, as it may cause unexpected failures due to the existing usage of application code in some of the already-released migrations.
```